### PR TITLE
Endpoint analysis diagram in mermaidjs

### LINKED ISSUE #433: Consider using https://mermaidjs.github.io/#/
Possible alternative to plantuml.

### PR DESCRIPTION
**Pull request by ashwinsajiv**
Fixes #433 

Changes proposed in this pull request:
- Sysl can now generate mermaidjs for endpoint analysis
- Also removed old mermaid directory

### CODE REVIEW

**Comment by ashwinsajiv** [comment]
@joshcarp - The test cases for all three diagrams will get more meaningful assertions added to it, this is just to make sure that any further changes to these files don't break the execution completely. I'll let you know when I change these test cases.

File: pkg/mermaid/endpointanalysisdiagram/endpointanalysisdiagram.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+//Accepts the sysl module and returns a string (and an error if any)
+//The resulting string is the mermaid code for the endpoint analysis for that application and endpoint
+func GenerateEndpointAnalysisDiagram(m *sysl.Module) (string, error) {
+	return generateEndpointAnalysisDiagram(m, &[]externalLink{}, true)
+}
+
+//This is a helper which has additional arguments which need not be entered by the user
+func generateEndpointAnalysisDiagram(m *sysl.Module, externalLinks *[]externalLink, theStart bool) (string, error) {
```
Comment (joshcarp) [inline pkg/mermaid/endpointanalysisdiagram/endpointanalysisdiagram.go:24]: FYI godoc style comments are `// <func name> does whatever`

File: pkg/mermaid/endpointanalysisdiagram/endpointanalysisdiagram.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	var result string
+	if theStart {
+		result = "%% AUTOGENERATED CODE -- DO NOT EDIT!\n\ngraph TD\n"
+	}
+	for appName, app := range m.Apps {
```
Comment (joshcarp) [inline pkg/mermaid/endpointanalysisdiagram/endpointanalysisdiagram.go:29]: Iterating over a map; does this not create non consistent mermaid code?
Maps don't ensure ordering, which is why sometimes you'll see this: https://github.com/joshcarp/sysl-printer/blob/master/alphabetical.go

File: pkg/mermaid/endpointanalysisdiagram/endpointanalysisdiagram.go (hunk 1 above)
Comment (ashwinsajiv) [reply pkg/mermaid/endpointanalysisdiagram/endpointanalysisdiagram.go:24]: Addressed

File: pkg/mermaid/endpointanalysisdiagram/endpointanalysisdiagram.go (hunk 2 above)
Comment (ashwinsajiv) [reply pkg/mermaid/endpointanalysisdiagram/endpointanalysisdiagram.go:29]: The order does not break the output or the code as long as all the apps are visited. Does the order of the output matter?

**Pull request merged**